AutoRest/Generators/Ruby/Ruby/TemplateModels/MethodTemplateModel.cs

```diff
@@ -30,6 +32,172 @@ public MethodTemplateModel(Method source, ServiceClient serviceClient)
+        /// Gets the return type name for the underlying interface method
+        /// </summary>
+        public virtual string OperationResponseReturnTypeString
+        {
+            get
+            {
+                return "MsRest::HttpOperationResponse";
+            }
+        }
+
+        /// <summary>
+        /// Gets the type for operation exception
+        /// </summary>
+        public virtual string OperationExceptionTypeString
+        {
+            get
+            {
+                return "MsRest::HttpOperationError";
+            }
+        }
+
+        /// <summary>
+        /// Gets the code required to initialize response body.
+        /// </summary>
+        public virtual string InitializeResponseBody
+        {
+            get { return string.Empty; }
+        }
+
+        /// <summary>
+        /// Gets the list of namespaces where we look for classes that need to
+        /// be instantiated dynamically due to polymorphism.
+        /// </summary>
+        public virtual List<string> ClassNamespaces
+        {
+            get
+            {
+                return new List<string> { };
+            }
+        }
+
+        /// <summary>
+        /// Gets the path parameters as a Ruby dictionary string
+        /// </summary>
+        public virtual string PathParamsRbDict
+        {
+            get
+            {
+                return ParamsToRubyDict(EncodingPathParams);
+            }
+        }
+
+        /// <summary>
+        /// Gets the skip encoding path parameters as a Ruby dictionary string
+        /// </summary>
+        public virtual string SkipEncodingPathParamsRbDict
+        {
+            get
+            {
+                return ParamsToRubyDict(SkipEncodingPathParams);
+            }
+        }
+
+        /// <summary>
+        /// Gets the query parameters as a Ruby dictionary string
+        /// </summary>
+        public virtual string QueryParamsRbDict
+        {
+            get
+            {
+                return ParamsToRubyDict(EncodingQueryParams);
+            }
+        }
+
+        /// <summary>
+        /// Gets the skip encoding query parameters as a Ruby dictionary string
+        /// </summary>
+        public virtual string SkipEncodingQueryParamsRbDict
+        {
+            get
+            {
+                return ParamsToRubyDict(SkipEncodingQueryParams);
+            }
+        }
+
+        /// <summary>
+        /// Gets the skip encoding path parameters
+        /// </summary>
+        public virtual IEnumerable<ParameterTemplateModel> SkipEncodingPathParams
+        {
+            get
+            {
+                return AllPathParams.Where(p => 
+                    (p.Extensions.ContainsKey(Generator.Extensions.SkipUrlEncodingExtension) &&
+                    String.Equals(p.Extensions[Generator.Extensions.SkipUrlEncodingExtension].ToString(), "true", StringComparison.OrdinalIgnoreCase) &&
+                    !p.Extensions.ContainsKey("hostParameter")));
```

Comment:
why do we condition on not containing key "hostParameter" here and not in the "EncodingPathParams" method?